From: Hariprasad Shenai <hariprasad@chelsio.com>
Date: Fri, 8 Jan 2016 12:21:25 +0530
Subject: cxgb4: Fixes static checker warning in mps_tcam_show()
Patch-mainline: v4.5-rc1
Git-commit: 89e7a154e01cffa325874e90cbb124057416de02
References: bsc#963896 FATE#320113

The commit 115b56af88b5 ("cxgb4: Update mps_tcam output to include T6
fields") from Dec 23, 2015, leads to the following static checker
warning:

        drivers/net/ethernet/chelsio/cxgb4/cxgb4_debugfs.c:1735
mps_tcam_show()
        warn: we tested 'lookup_type' before and it was 'true'

Fixing it.

Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/chelsio/cxgb4/cxgb4_debugfs.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_debugfs.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_debugfs.c
@@ -1726,13 +1726,13 @@ static int mps_tcam_show(struct seq_file
 				seq_printf(seq,
 					   "%3u %02x:%02x:%02x:%02x:%02x:%02x "
 					   "%012llx %06x %06x    -    -   %3c"
-					   "      %3c  %4x   "
+					   "      'I'  %4x   "
 					   "%3c   %#x%4u%4d", idx, addr[0],
 					   addr[1], addr[2], addr[3],
 					   addr[4], addr[5],
 					   (unsigned long long)mask,
 					   vniy, vnix, dip_hit ? 'Y' : 'N',
-					   lookup_type ? 'I' : 'O', port_num,
+					   port_num,
 					   (cls_lo & T6_SRAM_VLD_F) ? 'Y' : 'N',
 					   PORTMAP_G(cls_hi),
 					   T6_PF_G(cls_lo),
